pkg/registry/apps/statefulset/strategy.go

```diff
@@ -84,6 +104,12 @@ func (statefulSetStrategy) PrepareForUpdate(ctx context.Context, obj, old runtim
 
 	pod.DropDisabledTemplateFields(&newStatefulSet.Spec.Template, &oldStatefulSet.Spec.Template)
 
+	if !utilfeature.DefaultFeatureGate.Enabled(features.MaxUnavailableStatefulSet) && !maxUnavailableInUse(oldStatefulSet) {
+		if newStatefulSet.Spec.UpdateStrategy.RollingUpdate != nil {
+			newStatefulSet.Spec.UpdateStrategy.RollingUpdate.MaxUnavailable = nil
```

Comment:
you got the changes also in the wrong (non API commit) https://github.com/kubernetes/kubernetes/pull/82162/commits/6bb42dbd62f581c4af9af646211c28cb73157e8c#diff-453cdbb00bec23e49e2da14c1989f01ef2705560867257fe2630c3f516ec6404R91